compute_endpoint/globus_compute_endpoint/engines/globus_compute.py

```diff
@@ -288,13 +290,15 @@ def _submit(
         resource_specification: t.Dict,
         *args: t.Any,
         **kwargs: t.Any,
-    ) -> Future:
+    ) -> GCExecutorFuture:
         if resource_specification:
             raise ValueError(
                 "resource_specification is not supported on GlobusComputeEngine."
                 " For MPI apps, use GlobusMPIEngine."
             )
-        return self.executor.submit(func, resource_specification, *args, **kwargs)
+        f = self.executor.submit(func, resource_specification, *args, **kwargs)
```

Comment:
As we do in other engines, we should cast the returned `Future` to a `GCExecutorFuture`.
```suggestion
        f = t.cast(
            GCExecutorFuture,
            self.executor.submit(func, resource_specification, *args, **kwargs)
        )
```